test/sql/order_by_queries_sql_test.cpp

```diff
@@ -166,5 +163,74 @@ TEST_F(OrderByQueriesSQLTests, OrderByWithoutColumnsAndLimitTest) {
   auto txn = txn_manager.BeginTransaction();
   catalog::Catalog::GetInstance()->DropDatabaseWithName(DEFAULT_DB_NAME, txn);
   txn_manager.CommitTransaction(txn);
-}}  // namespace test
+}
+
+TEST_F(OrderByQueriesSQLTests, OrderByStar) {
+  catalog::Catalog::GetInstance()->CreateDatabase(DEFAULT_DB_NAME, nullptr);
+
+  CreateAndLoadTable();
+
+  std::vector<ResultType> result;
+  std::vector<FieldInfoType> tuple_descriptor;
+  std::string error_message;
+  int rows_changed;
+  std::unique_ptr<optimizer::AbstractOptimizer> optimizer(
+      new optimizer::SimpleOptimizer());
+
+  std::string query("SELECT * FROM test ORDER BY d");
+  auto select_plan = SQLTestsUtil::GeneratePlanWithOptimizer(optimizer, query);
+  EXPECT_EQ(select_plan->GetPlanNodeType(), PLAN_NODE_TYPE_ORDERBY);
+
+  SQLTestsUtil::ExecuteSQLQueryWithOptimizer(
+      optimizer, query, result, tuple_descriptor, rows_changed, error_message);
+  // Check the return value
+  // Should be: [1, 22, 333, 'abcd']; [3,...], [2,...];
+  EXPECT_EQ(0, rows_changed);
+  EXPECT_EQ("1", SQLTestsUtil::GetResultValueAsString(result, 0));
+  EXPECT_EQ("22", SQLTestsUtil::GetResultValueAsString(result, 1));
+  EXPECT_EQ("333", SQLTestsUtil::GetResultValueAsString(result, 2));
+  EXPECT_EQ("abcd", SQLTestsUtil::GetResultValueAsString(result, 3));
+  EXPECT_EQ("3", SQLTestsUtil::GetResultValueAsString(result, 4));
+  EXPECT_EQ("2", SQLTestsUtil::GetResultValueAsString(result, 8));
+
+  // free the database just created
+  auto &txn_manager = concurrency::TransactionManagerFactory::GetInstance();
+  auto txn = txn_manager.BeginTransaction();
+  catalog::Catalog::GetInstance()->DropDatabaseWithName(DEFAULT_DB_NAME, txn);
+  txn_manager.CommitTransaction(txn);
+}
+
+TEST_F(OrderByQueriesSQLTests, OrderByStarWithLimit) {
+  catalog::Catalog::GetInstance()->CreateDatabase(DEFAULT_DB_NAME, nullptr);
+
+  CreateAndLoadTable();
+
+  std::vector<ResultType> result;
+  std::vector<FieldInfoType> tuple_descriptor;
+  std::string error_message;
+  int rows_changed;
+  std::unique_ptr<optimizer::AbstractOptimizer> optimizer(
+      new optimizer::SimpleOptimizer());
+
+  std::string query("SELECT * FROM test ORDER BY d LIMIT 2");
+  auto select_plan = SQLTestsUtil::GeneratePlanWithOptimizer(optimizer, query);
+  EXPECT_EQ(select_plan->GetPlanNodeType(), PLAN_NODE_TYPE_LIMIT);
+  EXPECT_EQ(select_plan->GetChildren()[0]->GetPlanNodeType(), PLAN_NODE_TYPE_ORDERBY);
+
+  SQLTestsUtil::ExecuteSQLQueryWithOptimizer(
+      optimizer, query, result, tuple_descriptor, rows_changed, error_message);
+  // Check the return value
+  // Should be: 1, 3
+  EXPECT_EQ(0, rows_changed);
+  EXPECT_EQ("1", SQLTestsUtil::GetResultValueAsString(result, 0));
+  EXPECT_EQ("3", SQLTestsUtil::GetResultValueAsString(result, 4));
+
```

Comment:
Got it, I'll add those as soon as I get home. I saw the coveralls report and figured that there are no `DESC` test as well. I'll also add that.